pandas/tests/io/test_s3.py

```diff
@@ -1,8 +1,28 @@
-from pandas.io.common import _is_s3_url
+import pytest
+import pandas as pd
 
+from pandas.io.common import _is_s3_url
+from pandas.util import testing as tm
 
 class TestS3URL(object):
 
     def test_is_s3_url(self):
         assert _is_s3_url("s3://pandas/somethingelse.com")
         assert not _is_s3_url("s4://pandas/somethingelse.com")
+
+class TestIntegration(object):
```

Comment:
Move this to `pandas/tests/io/test_parquet.py` and make a new class `TestS3`, that's marked with `tm.network`.

Then your test methods should take an argument `s3_resource` to use that fixture: https://github.com/pandas-dev/pandas/blob/c753e1e08b01a438aaa252327de046109bf4bcfd/pandas/tests/io/conftest.py#L29

That will take care of all the skipping / mocking for you. You just have to write the test at that point.